core/src/main/java/org/springframework/security/authentication/ott/OneTimeTokenUtils.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authentication.ott;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.UUID;
+
+import org.springframework.util.Assert;
+
+/**
+ *
+ * Utility for default generation and checking of {@link OneTimeToken} time to live.
+ *
+ * @author Max Batischev
+ * @since 6.4
+ */
+public final class OneTimeTokenUtils {
+
+	public static long DEFAULT_ONE_TIME_TOKEN_TIME_TO_LIVE = 300;
+
+	private static Clock clock = Clock.systemUTC();
```

Comment:
I don't view it as ideal to have a static Clock for a few reasons.

It makes it easier to have errors when setting it. 

1) For example, the tests right now only set this before a test is ran. However, some tests override the value. All future tests that use the Clock must also set it. Alternatively, we could also set after the tests run
2) The tests that are setting the value must know what to set the Clock to. Any changes to the Clock would likely need to be made in all of the tests.

Ensuring we are using an instance variable (perhaps on the Repository itself) would avoid all of this. It would also allow us to potentially expose the Clock on the Repository to be set in the future.

Reply:
Yes, you are right. Not the best solution. I will return the logic for generating and checking the token from the `InMemoryOneTimeTokenService`, I will need to think about how to extract it in the next iterations.